scripts/examples/str_comp_func/test_str_comp_func.py

```diff
@@ -0,0 +1,95 @@
+# tests based on string comparisons with boolean return values
+# one input parameter <param> of type str
+
+# 1. if <param> == "test123"
+def test_eq(param: str) -> bool:    
+    if param == "test123":
+        return True
+    return False
+
+# 2. if "test123" == <param>
+def test_eq_rev(param: str) -> bool:    
+    if "test123" == param:
+        return True
+    return False
+
+# 3. if "test123"       == <param>  (extra spaces)
+def test_eq_spaces(param: str) -> bool:    
+    if "test123"       == param:
+        return True
+    return False
+
+# 4. if "test123"==<param>  (no spaces)
+def test_eq_no_spaces(param: str) -> bool:    
+    if "test123"==param:
+        return True
+    return False
+
+# 5. contradiction
+def test_contradiction(param: str) -> bool:
+    if param == "arbok" and param == "kobra":
+        return True
+    return False
+
+# 6. if <param> == "hello" or <param> == "world"
+def test_eq_or(param: str) -> bool:    
+    if param == "hello" or param == "world":
+        return True
+    return False
+
+# 7. two if-statements
+def test_two_ifs(param: str) -> bool:
+    if param == "test123":
```

Comment:
Is it intended that those if-statements are contradicting?

If so, we should also have a test with two if-statements below each other and not nested.